packages/next-plugin/src/index.js

```diff
@@ -0,0 +1,34 @@
+import { VanillaExtractPlugin } from '@vanilla-extract/webpack-plugin';
+import { getGlobalCssLoader } from 'next/dist/build/webpack/config/blocks/css/loaders';
+
+export const createVanillaExtractPlugin =
+  (pluginOptions = {}) =>
+  (nextConfig = {}) =>
+    Object.assign({}, nextConfig, {
+      webpack(config, options) {
+        const { dev, isServer } = options;
+
+        config.module.rules[config.module.rules.length - 1].oneOf.unshift({
```

Comment:
i think  there's no guarantee the css related rules are last because other plugins might have pushed to `config.module.rules`

Reply:
Yep, that's a good point. I'll see it I can figure out a good way to detect which item in the array should be modified.

Reply:
maybe something like `config.module.rules.find(rule => Array.isArray(rule.oneOf) && rule.oneOf.some(r => r.test.test('test.css')))`

Reply:
Yep, I was thinking something like this.

Reply:
Pushed an update to make this work similar to how you suggested. Thanks for the feedback 👍